include/nbl/builtin/glsl/utils/swapchain_transform.glsl

```diff
@@ -0,0 +1,146 @@
+        return coord;
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_90:
+        return ivec2(coord.y, lastTexel.x - coord.x);
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_180:
+        return ivec2(lastTexel) - coord;
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_270:
+        return ivec2(lastTexel.y - coord.y, coord.x);
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR:
+        return ivec2(lastTexel.x - coord.x, coord.y);
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_90:
+        return coord.yx;
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_180:
+        return ivec2(coord.x, lastTexel.y - coord.y);
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_270:
+        return lastTexel - coord.yx;
+    default:
+        return ivec2(0);
+    }
+}
+
+ivec2 nbl_glsl_swapchain_transform_backward(in uint swapchainTransform, in ivec2 coord, in ivec2 lastTexel) {
+    switch (swapchainTransform) 
+    {
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_IDENTITY:
+        return coord;
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_90:
+        return ivec2(lastTexel.y - coord.y, coord.x);
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_180:
+        return ivec2(lastTexel) - coord;
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_ROTATE_270:
+        return ivec2(coord.y, lastTexel.x - coord.x);
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR:
+        return ivec2(lastTexel.x - coord.x, coord.y);
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_90:
+        return lastTexel - coord.yx;
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_180:
+        return ivec2(coord.x, lastTexel.y - coord.y);
+    case NBL_GLSL_SWAPCHAIN_TRANSFORM_E_HORIZONTAL_MIRROR_ROTATE_270:
+        return coord.yx;
+    default:
+        return ivec2(0);
+    }
+}
+
+// If using `imageStore` to write *directly to the swapchain*, you may use this function to 
+// transform the UVs before rendering with them (e.g. path tracer that will use UVs to fire rays, gather). 
+// Remember to only use one transform in the pipeline!
+ivec2 nbl_glsl_swapchain_transform_preTransform(in uint swapchainTransform, in ivec2 coord, in ivec2 screenSize) {
+    return nbl_glsl_swapchain_transform_backward(swapchainTransform, coord, screenSize - 1);
+}
+
+// If using `imageStore` to write *directly to the swapchain*, you may use this function to 
+// transform the coordinates written to the swapchain (the ones used in imageStore, scatter).
+// Remember to only use one transform in the pipeline!
+ivec2 nbl_glsl_swapchain_transform_postTransform(in uint swapchainTransform, in ivec2 coord, in ivec2 screenSize) {
+    return nbl_glsl_swapchain_transform_forward(swapchainTransform, coord, screenSize - 1);
+}
+
+// If rendering directly to the swapchain, use the transformed extents rather than the actual
+// extents for aspect ratio calculations, in order to be consistent with 90º/270º rotations.
```

Comment:
//! [width,height]` might switch to [height, width] in orientations such as 90°CW
// Usecase: Currently none in the shaders